turbo/transactions/call.go

```diff
@@ -124,3 +126,99 @@ func getHashGetter(requireCanonical bool, tx kv.Tx, headerReader services.Header
 		return h.Hash()
 	}
 }
+
+type ReusableCaller struct {
+	evm         *vm.EVM
+	gasCap      uint64
+	baseFee     *uint256.Int
+	stateReader state.StateReader
+}
+
+func (r *ReusableCaller) DoCall(
+	ctx context.Context,
+	args ethapi2.CallArgs,
+	callTimeout time.Duration,
+) (*core.ExecutionResult, error) {
+	var cancel context.CancelFunc
+	if callTimeout > 0 {
+		ctx, cancel = context.WithTimeout(ctx, callTimeout)
+	} else {
+		ctx, cancel = context.WithCancel(ctx)
+	}
+
+	// Make sure the context is cancelled when the call has completed
+	// this makes sure resources are cleaned up.
+	defer cancel()
+
+	msg, err := args.ToMessage(r.gasCap, r.baseFee)
+	if err != nil {
+		return nil, err
+	}
+
+	// reset the EVM so that we can continue to use it with the new context
+	txCtx := core.NewEVMTxContext(msg)
+	r.evm.Reset(txCtx, r.evm.IntraBlockState())
```

Comment:
looks weird to pass self-parameter

Reply:
It does, but I got nil pointer exceptions when trying to new up a new state, will take another look